src/ChargeDrift/ChargeDrift.jl

```diff
@@ -80,12 +80,8 @@ function modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv
 end
 modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv::Missing) where {T} = sv
 
-@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cylindrical})::Bool where {T <: SSDFloat}
-    pt_cyl::CylindricalPoint{T} = CylindricalPoint(pt)
-    pt_cyl in point_types 
-end
-@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cartesian})::Bool where {T <: SSDFloat}
-    pt in point_types 
+@inline function _is_next_point_in_det(pt::AbstractCoordinatePoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, S})::Bool where {T <: SSDFloat, S}
+    pt in point_types || (pt in det.semiconductor && !(pt in det.contacts))

```

Comment:
You need to convert the point to the right coordinates first before you check if it's in `point_types`.
For checking geometries, this should be fine as it is.

Maybe something like:
```julia
@inline function _is_next_point_in_det(pt::AbstractCoordinatePoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, S})::Bool where {T <: SSDFloat, S}
    _convert_point(pt, S) in point_types || (pt in det.semiconductor && !(pt in det.contacts))
end